tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,544 @@
+                    {
+                        _skipTests = true;
+                        _skipReason = $"Missing required environment variable: {ResourceGroupEnvVar}";
+                        return;
+                    }
+
+                    _testResourceGroupName = resourceGroupName;
+                    _resourceGroup = await subscription.GetResourceGroups().GetAsync(resourceGroupName);
+                }
+
+                // Create unique store name for this test run
+                _testStoreName = $"integration-{Guid.NewGuid():N}".Substring(0, 20);
+
+                // Create the App Configuration store
+                var storeData = new AppConfigurationStoreData(new AzureLocation(location), new AppConfigurationSku("free"));
+                var createOperation = await _resourceGroup.GetAppConfigurationStores().CreateOrUpdateAsync(
+                    WaitUntil.Completed,
+                    _testStoreName,
+                    storeData);
+
+                _appConfigStore = createOperation.Value;
+                _appConfigEndpoint = new Uri(_appConfigStore.Data.Endpoint);
+
+                // Get the connection string for the store instead of using RBAC
+                var accessKeys = _appConfigStore.GetKeysAsync();
+                var primaryKey = await accessKeys.FirstOrDefaultAsync();
+
+                if (primaryKey == null)
+                {
+                    throw new InvalidOperationException("Failed to retrieve access keys from App Configuration store.");
+                }
+
+                _connectionString = primaryKey.ConnectionString;
+
+                // Initialize the configuration client with the connection string
+                _configClient = new ConfigurationClient(_connectionString);
+
+                // Add test settings to the store
+                foreach (var setting in _testSettings)
+                {
+                    await _configClient.SetConfigurationSettingAsync(setting);
+                }
+            }
+            catch (Exception ex)
+            {
+                _skipTests = true;
+                _skipReason = $"Failed to initialize integration tests: {ex.Message}";
+
+                // Clean up any partially created resources
+                await DisposeAsync();
+            }
+        }
+
+        /// <summary>
+        /// Cleans up the temporary App Configuration store after tests are complete.
+        /// </summary>
+        public async Task DisposeAsync()
+        {
+            // Don't attempt cleanup if we don't have a store to delete
+            if (_appConfigStore == null && !_shouldDeleteResourceGroup)
+            {
+                return;
+            }
+
+            try
+            {
+                if (_appConfigStore != null)
+                {
+                    await _appConfigStore.DeleteAsync(WaitUntil.Completed);
+                }
+
+                if (_shouldDeleteResourceGroup && _resourceGroup != null)
```

Comment:
Do you think we'll ever want to delete the resource group?

Reply:
If we're creating it for the purpose of the test, I assumed we delete it along with the store. If there's a different to get a resource group or have whoever calls the test pass one in, then no.